packages/core/src/styles/Modal.scss

```diff
@@ -107,3 +107,38 @@ $component: #{$prefix}-modal;
         }
     }
 }
+
+// -------------------------------------
+//   width indent for multiple modal
+// -------------------------------------
+
+@for $i from 1 through $modal-indent-layer-max - 1 {
+    // ---------------------------------
+    // this will generate selector like `.#{$prefix}-base-layer ~ .#{$prefix}-base-layer .#{$prefix-modal}`
+    // which is long and ugly, but
+    // 1. it's more complicated and error prone to handle layer count in React side
+    // 2. it's rare to use multiple modal in our use cases.
+    // so we handle indent by css.
+    // ---------------------------------
+    #{repeat-str('.#{$prefix}-base-layer ~', $i)} .#{$prefix}-base-layer {
```

Comment:
I'm not sure what's the best way to control the indent effect. Discussion welcome :-)

Currently I choose hardcode the width and height of `gyp-base-layer` in css since when there are more than 7 modals we won't indent the modal. It's simple. But the generated css rule will be long and ugly in developer tool (`.gyp-base-layer ~ .gyp-base-layer ~ .gyp-base-layer ~ ...`). But I couldn't figure out a more concise way to express this.

I have also considered tracking the layer number in javascript so we can add class like `.gyp-modal-3` to control width. But I think it's more complicated and error prone to manage layer number in `renderToLayer` so I don't take this way.

Reply:
I think it may be the current best solution to this problem. Since it is indeed more error-prone to handle this problem using Javascript, and `.gyp-base-layer` layers are inserted in <body>, it is impossible to use selectors like `nth-of-type` to solve this problem. I think `.gyp-base-layer ~ .gyp-base-layer ~ .gyp-base-layer ~ ...` should be a great solution.

What does @zhusee2 think?

Reply:
I think it's possible to use `:nth-of-type()` to make this even easier. But that needs a try.
Consider tweaking `renderToLayer()` HOC to include `ComponentName` in its layer classname, so the layers are more descriptive and easier to select:

```html
<div class="gyp-base-layer modal" />
<div class="gyp-base-layer popover" />
<div class="gyp-base-layer modal" />
```
Then you should be able to apply different width or margin using `.gyp-base-layer.modal:nth-of-type()`?

Reply:
I think it's a great solution because it can achieve similar things without having to introduce complex JavaScript logic. Let's give it a shot!